packages/tale_knowledge/src/tale_knowledge/extraction/pptx.py

```
@@ -0,0 +1,152 @@
+        Tuple of (extracted_text, vision_was_used).
+    """
+    logger.info(f"Processing PPTX: {filename}")
+
+    try:
+        zf = zipfile.ZipFile(BytesIO(pptx_bytes))
+        total = sum(info.file_size for info in zf.infolist())
+        if total > MAX_UNCOMPRESSED_SIZE:
+            raise ValueError(f"File exceeds maximum decompressed size ({total} bytes)")
+        zf.close()
+    except zipfile.BadZipFile:
+        raise ValueError("Invalid or corrupt file")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)